### PR TITLE
Dedicate public method for fixers by name

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Permit to get fixers by name and do some checks manually.

Concrete case: https://github.com/Soullivaneuh/php-cs-fixer-styleci-bridge/pull/21

### CODE REVIEW

**Comment by keradus** [comment]
Don't like the idea.
`getFixers` return fixers in good order - and getting fixers in other order create possibility to use them in wrong order, which will lead to errors.

I would suggest other method like `hasRule(string)`.

Also, when adding new public method remember to add tests for it.

**Comment by Soullivaneuh** [comment]
I would suggest other method like `hasRule(string)`.

Why not. But in this case we should have `$fixersAsName` as a property and `getFixersAsName` private because this will be used twice. Do you agree?

**Comment by keradus** [comment]
We could have `$fixersAsName` or not, it's not part of interface but implementation. If it would made implementation better why not.

**Comment by Soullivaneuh** [comment]
Closed in favor of #1432.

File: Symfony/CS/FixerFactory.php [hunk 1]
```diff
@@ -139,7 +155,7 @@ public function registerFixer(FixerInterface $fixer)
      */
     public function useRuleSet(RuleSetInterface $ruleSet)
     {
-        $fixersByName = array();
+        $fixersByName = $this->getFixersByName();
 
         foreach ($this->fixers as $fixer) {
```
Comment (GrahamCampbell) [inline Symfony/CS/FixerFactory.php]: why is this code running twice?

**Pull request closed without merging**